data_model/src/events/data.rs

```diff
@@ -0,0 +1,498 @@
+            .status
+            .map_or(true, |status| status.apply(event.status));
+        entity_check && status_check
+    }
+}
+
+impl EntityFilter {
+    fn apply(&self, entity: &Entity) -> bool {
+        match self {
+            Self::Account(opt) => match entity {
+                Entity::Account(account_id) => opt
+                    .as_ref()
+                    .map_or(true, |filter_id| account_id == filter_id),
+                Entity::Asset(asset_id) => opt
+                    .as_ref()
+                    .map_or(false, |filter_id| asset_id.account_id == *filter_id),
+                _ => false,
+            },
+            Self::AssetDefinition(opt) => match entity {
+                Entity::AssetDefinition(asset_definition_id) => opt
+                    .as_ref()
+                    .map_or(true, |filter_id| asset_definition_id == filter_id),
+                Entity::Asset(asset_id) => opt
+                    .as_ref()
+                    .map_or(false, |filter_id| asset_id.definition_id == *filter_id),
+                _ => false,
+            },
+            Self::Asset(opt) => match entity {
+                Entity::Asset(asset_id) => {
+                    opt.as_ref().map_or(true, |filter_id| asset_id == filter_id)
+                }
+                _ => false,
+            },
+            Self::Domain(opt) => match entity {
+                Entity::Account(account_id) => opt
+                    .as_ref()
+                    .map_or(false, |filter_id| account_id.domain_id == *filter_id),
+                Entity::AssetDefinition(asset_definition_id) => {
+                    opt.as_ref().map_or(false, |filter_id| {
+                        asset_definition_id.domain_id == *filter_id
+                    })
+                }
+                Entity::Asset(asset_id) => opt.as_ref().map_or(false, |filter_id| {
+                    asset_id.account_id.domain_id == *filter_id
+                        || asset_id.definition_id.domain_id == *filter_id
+                }),
+                Entity::Domain(id) => opt.as_ref().map_or(true, |filter_id| id == filter_id),
+                _ => false,
+            },
+            Self::Peer(opt) => match entity {
+                Entity::Peer(peer_id) => {
+                    opt.as_ref().map_or(true, |filter_id| peer_id == filter_id)
+                }
+                _ => false,
+            },
+        }
+    }
+}
+
+impl StatusFilter {
+    fn apply(self, status: Status) -> bool {
+        match self {
+            Self::Created => Status::Created == status,
+            Self::Updated(opt) => match status {
+                Status::Updated(detail) => {
+                    opt.map_or(true, |filter_detail| detail == filter_detail)
+                }
+                _ => false,
+            },
+            Self::Deleted => Status::Deleted == status,
+        }
+    }
+}
+
+impl From<Event> for Vec<Event> {
+    fn from(src: Event) -> Self {
+        vec![src]
+    }
+}
```

Comment:
this doesn't seem to bring much? I'd rather not have it

Reply:
This impl allows not a few instruction executions have their `type Diff` single `DataEvent`,
which is clearer than `Vec<Event>`